### ISSUE TITLE
extra `-` in deleted file

### ISSUE THREAD

**Issue by stevemao** [so-fancy/diff-so-fancy#201]
Hi, I got this minor problem in the latest version. The rest of the diff looks fine.

<img width="647" alt="screen shot 2016-09-25 at 2 40 01 pm" src="https://cloud.githubusercontent.com/assets/6316590/18813079/2d4ffecc-832e-11e6-98fe-7f3896ccf432.png">

Z2l0IGRpZmYgSEVBRF4KZGlmZiAtLWdpdCB0ZXN0L3JlYWR5d2FpdC5odG1sIHRlc3QvZGF0YS9yZWFkeXdhaXQuaHRtbApzaW1pbGFyaXR5IGluZGV4IDY0JQpyZW5hbWUgZnJvbSB0ZXN0L3JlYWR5d2FpdC5odG1sCnJlbmFtZSB0byB0ZXN0L2RhdGEvcmVhZHl3YWl0Lmh0bWwKaW5kZXggZTM0YTBkNy4uYWNjOWRkNSAxMDA2NDQKLS0tIHRlc3QvcmVhZHl3YWl0Lmh0bWwKKysrIHRlc3QvZGF0YS9yZWFkeXdhaXQuaHRtbApAQCAtMTQsMjEgKzE0LDQ0IEBACiAJCSNvdXRwdXQgeyBiYWNrZ3JvdW5kLWNvbG9yOiBncmVlbiB9CiAJCSNleHBlY3RlZE91dHB1dCB7IGJhY2tncm91bmQtY29sb3I6IGdyZWVuIH0KIAk8L3N0eWxlPgotCTxzY3JpcHQgc3JjPSJqcXVlcnkuanMiPjwvc2NyaXB0PgorCTxzY3JpcHQgc3JjPSIuLi9qcXVlcnkuanMiPjwvc2NyaXB0PgorCTxzY3JpcHQgc3JjPSIuL2lmcmFtZVRlc3QuanMiPjwvc2NyaXB0PgogCiAJPCEtLSBMb2FkIHRoZSBzY3JpcHQgbG9hZGVyIHRoYXQgdXNlcwogCQlqUXVlcnkucmVhZHlXYWl0IC0tPgotCTxzY3JpcHQgc3JjPSJkYXRhL3JlYWR5d2FpdGxvYWRlci5qcyI+PC9zY3JpcHQ+CiAKIAk8c2NyaXB0IHR5cGU9InRleHQvamF2YXNjcmlwdCI+CisJLy8gSG9sZCBvbiBqUXVlcnkhCisJalF1ZXJ5LmhvbGRSZWFkeSggdHJ1ZSApOworCisJdmFyIHJlYWR5UmVnRXhwID0gL14oY29tcGxldGV8bG9hZGVkKSQvOworCisJZnVuY3Rpb24gYXNzZXRMb2FkZWQoIGV2dCApIHsKKwkJdmFyIG5vZGUgPSBldnQuY3VycmVudFRhcmdldCB8fCBldnQuc3JjRWxlbWVudDsKKwkJaWYgKCBldnQudHlwZSA9PT0gImxvYWQiIHx8IHJlYWR5UmVnRXhwLnRlc3QoIG5vZGUucmVhZHlTdGF0ZSApICkgeworCQkJalF1ZXJ5LmhvbGRSZWFkeSggZmFsc2UgKTsKKwkJfQorCX0KKworCXNldFRpbWVvdXQoIGZ1bmN0aW9uKCkgeworICAgIHZhciBzY3JpcHQgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCAic2NyaXB0IiApOworICAgIHNjcmlwdC50eXBlID0gInRleHQvamF2YXNjcmlwdCI7CisgICAgaWYgKCBzY3JpcHQuYWRkRXZlbnRMaXN0ZW5lciApIHsKKyAgICAgICAgc2NyaXB0LmFkZEV2ZW50TGlzdGVuZXIoICJsb2FkIiwgYXNzZXRMb2FkZWQsIGZhbHNlICk7CisgICAgfSBlbHNlIHsKKyAgICAgICAgc2NyaXB0LmF0dGFjaEV2ZW50KCAib25yZWFkeXN0YXRlY2hhbmdlIiwgYXNzZXRMb2FkZWQgKTsKKyAgICB9CisgICAgc2NyaXB0LnNyYyA9ICIuL3JlYWR5d2FpdGFzc2V0LmpzIjsKKyAgICBkb2N1bWVudC5nZXRFbGVtZW50c0J5VGFnTmFtZSggImhlYWQiIClbIDAgXS5hcHBlbmRDaGlsZCggc2NyaXB0ICk7CisJfSwgMjAwMCApOworCiAJalF1ZXJ5KGZ1bmN0aW9uKCkgewotCQkvLyBUaGUgZGVsYXllZE1lc3NhZ2UgaXMgZGVmaW5lZCBieQotCQkvLyB0aGUgcmVhZHl3YWl0YXNzZXQuanMgZmlsZSwgc28gdGhlCi0JCS8vIG5leHQgbGluZSB3aWxsIG9ubHkgd29yayBpZiB0aGlzIERPTQorCQkvLyBUaGUgbmV4dCBsaW5lIHdpbGwgb25seSB3b3JrIGlmIHRoaXMgRE9NCiAJCS8vIHJlYWR5IGNhbGxiYWNrIGlzIGNhbGxlZCBhZnRlciByZWFkeVdhaXQKIAkJLy8gaGFzIGJlZW4gZGVjcmVtZW50ZWQgYnkgcmVhZHl3YWl0bG9hZGVyLmpzCiAJCS8vIElmIGFuIGVycm9yIG9jY3Vycy4KIAkJalF1ZXJ5KCIjb3V0cHV0IikuYXBwZW5kKGRlbGF5ZWRNZXNzYWdlKTsKKwkJc3RhcnRJZnJhbWVUZXN0KCBkZWxheWVkTWVzc2FnZSApOwogCX0pOwogCTwvc2NyaXB0PgogPC9oZWFkPgpkaWZmIC0tZ2l0IHRlc3QvZGF0YS9yZWFkeXdhaXRhc3NldC5qcyB0ZXN0L2RhdGEvcmVhZHl3YWl0YXNzZXQuanMKZGVsZXRlZCBmaWxlIG1vZGUgMTAwNjQ0CmluZGV4IDIzMDg5NjUuLjAwMDAwMDAKLS0tIHRlc3QvZGF0YS9yZWFkeXdhaXRhc3NldC5qcworKysgL2Rldi9udWxsCkBAIC0xICswLDAgQEAKLXZhciBkZWxheWVkTWVzc2FnZSA9ICJJdCB3b3JrZWQhIjsKZGlmZiAtLWdpdCB0ZXN0L2RhdGEvcmVhZHl3YWl0bG9hZGVyLmpzIHRlc3QvZGF0YS9yZWFkeXdhaXRsb2FkZXIuanMKZGVsZXRlZCBmaWxlIG1vZGUgMTAwNjQ0CmluZGV4IDhmNGEzNDUuLjAwMDAwMDAKLS0tIHRlc3QvZGF0YS9yZWFkeXdhaXRsb2FkZXIuanMKKysrIC9kZXYvbnVsbApAQCAtMSwyNSArMCwwIEBACi0vLyBTaW1wbGUgc2NyaXB0IGxvYWRlciB0aGF0IHVzZXMgalF1ZXJ5LnJlYWR5V2FpdCB2aWEgalF1ZXJ5LmhvbGRSZWFkeSgpCi0KLS8vSG9sZCBvbiBqUXVlcnkhCi1qUXVlcnkuaG9sZFJlYWR5KCB0cnVlICk7Ci0KLXZhciByZWFkeVJlZ0V4cCA9IC9eKGNvbXBsZXRlfGxvYWRlZCkkLzsKLQotZnVuY3Rpb24gYXNzZXRMb2FkZWQoIGV2dCApIHsKLQl2YXIgbm9kZSA9IGV2dC5jdXJyZW50VGFyZ2V0IHx8IGV2dC5zcmNFbGVtZW50OwotCWlmICggZXZ0LnR5cGUgPT09ICJsb2FkIiB8fCByZWFkeVJlZ0V4cC50ZXN0KCBub2RlLnJlYWR5U3RhdGUgKSApIHsKLQkJalF1ZXJ5LmhvbGRSZWFkeSggZmFsc2UgKTsKLQl9Ci19Ci0KLXNldFRpbWVvdXQoIGZ1bmN0aW9uKCkgewotCXZhciBzY3JpcHQgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCAic2NyaXB0IiApOwotCXNjcmlwdC50eXBlID0gInRleHQvamF2YXNjcmlwdCI7Ci0JaWYgKCBzY3JpcHQuYWRkRXZlbnRMaXN0ZW5lciApIHsKLQkJc2NyaXB0LmFkZEV2ZW50TGlzdGVuZXIoICJsb2FkIiwgYXNzZXRMb2FkZWQsIGZhbHNlICk7Ci0JfSBlbHNlIHsKLQkJc2NyaXB0LmF0dGFjaEV2ZW50KCAib25yZWFkeXN0YXRlY2hhbmdlIiwgYXNzZXRMb2FkZWQgKTsKLQl9Ci0Jc2NyaXB0LnNyYyA9ICJkYXRhL3JlYWR5d2FpdGFzc2V0LmpzIjsKLQlkb2N1bWVudC5nZXRFbGVtZW50c0J5VGFnTmFtZSggImhlYWQiIClbIDAgXS5hcHBlbmRDaGlsZCggc2NyaXB0ICk7Ci19LCAyMDAwICk7CmRpZmYgLS1naXQgdGVzdC9kYXRhL3Rlc3Rpbml0LmpzIHRlc3QvZGF0YS90ZXN0aW5pdC5qcwppbmRleCBlZjIxMGU3Li5lMDY4NThkIDEwMDY0NAotLS0gdGVzdC9kYXRhL3Rlc3Rpbml0LmpzCisrKyB0ZXN0L2RhdGEvdGVzdGluaXQuanMKQEAgLTI3Niw2ICsyNzYsNyBAQCB0aGlzLmxvYWRUZXN0cyA9IGZ1bmN0aW9uKCkgewogCQkJCSJ1bml0L2Jhc2ljLmpzIiwKIAogCQkJCSJ1bml0L2NvcmUuanMiLAorCQkJCSJ1bml0L3JlYWR5d2FpdC5qcyIsCiAJCQkJInVuaXQvY2FsbGJhY2tzLmpzIiwKIAkJCQkidW5pdC9kZWZlcnJlZC5qcyIsCiAJCQkJInVuaXQvZGVwcmVjYXRlZC5qcyIsCmRpZmYgLS1naXQgdGVzdC91bml0L3JlYWR5d2FpdC5qcyB0ZXN0L3VuaXQvcmVhZHl3YWl0LmpzCm5ldyBmaWxlIG1vZGUgMTAwNjQ0CmluZGV4IDAwMDAwMDAuLmQ1NzJkMTAKLS0tIC9kZXYvbnVsbAorKysgdGVzdC91bml0L3JlYWR5d2FpdC5qcwpAQCAtMCwwICsxLDEwIEBACisvLyBTaW1wbGUgc2NyaXB0IGxvYWRlciB0aGF0IHVzZXMgalF1ZXJ5LnJlYWR5V2FpdCB2aWEgalF1ZXJ5LmhvbGRSZWFkeSgpCisKK3Rlc3RJZnJhbWUoCisJImhvbGRSZWFkeSB0ZXN0IG5lZWRzIHRvIGJlIGEgc3RhbmRhbG9uZSB0ZXN0IHNpbmNlIGl0IGRlYWxzIHdpdGggRE9NIHJlYWR5IiwKKwkicmVhZHl3YWl0Lmh0bWwiLAorCWZ1bmN0aW9uKCBhc3NlcnQsIGpRdWVyeSwgd2luZG93LCBkb2N1bWVudCwgZGVsYXllZE1lc3NhZ2UgKSB7CisJCWFzc2VydC5leHBlY3QoIDEgKTsKKwkJYXNzZXJ0Lm9rKCBkZWxheWVkTWVzc2FnZSwgIkl0ICIgKTsKKwl9CispOwoKCgpnaXQgZGlmZiBIRUFEXiAtLWNvbG9yChtbMTszM21kaWZmIC0tZ2l0IHRlc3QvcmVhZHl3YWl0Lmh0bWwgdGVzdC9kYXRhL3JlYWR5d2FpdC5odG1sG1ttChtbMTszM21zaW1pbGFyaXR5IGluZGV4IDY0JRtbbQobWzE7MzNtcmVuYW1lIGZyb20gdGVzdC9yZWFkeXdhaXQuaHRtbBtbbQobWzE7MzNtcmVuYW1lIHRvIHRlc3QvZGF0YS9yZWFkeXdhaXQuaHRtbBtbbQobWzE7MzNtaW5kZXggZTM0YTBkNy4uYWNjOWRkNSAxMDA2NDQbW20KG1sxOzMzbS0tLSB0ZXN0L3JlYWR5d2FpdC5odG1sG1ttChtbMTszM20rKysgdGVzdC9kYXRhL3JlYWR5d2FpdC5odG1sG1ttChtbMTszNW1AQCAtMTQsMjEgKzE0LDQ0IEBAG1ttCiAJCSNvdXRwdXQgeyBiYWNrZ3JvdW5kLWNvbG9yOiBncmVlbiB9G1ttCiAJCSNleHBlY3RlZE91dHB1dCB7IGJhY2tncm91bmQtY29sb3I6IGdyZWVuIH0bW20KIAk8L3N0eWxlPhtbbQobWzMxbS0JPHNjcmlwdCBzcmM9ImpxdWVyeS5qcyI+PC9zY3JpcHQ+G1ttChtbMzJtKxtbbQkbWzMybTxzY3JpcHQgc3JjPSIuLi9qcXVlcnkuanMiPjwvc2NyaXB0PhtbbQobWzMybSsbW20JG1szMm08c2NyaXB0IHNyYz0iLi9pZnJhbWVUZXN0LmpzIj48L3NjcmlwdD4bW20KIBtbbQogCTwhLS0gTG9hZCB0aGUgc2NyaXB0IGxvYWRlciB0aGF0IHVzZXMbW20KIAkJalF1ZXJ5LnJlYWR5V2FpdCAtLT4bW20KG1szMW0tCTxzY3JpcHQgc3JjPSJkYXRhL3JlYWR5d2FpdGxvYWRlci5qcyI+PC9zY3JpcHQ+G1ttCiAbW20KIAk8c2NyaXB0IHR5cGU9InRleHQvamF2YXNjcmlwdCI+G1ttChtbMzJtKxtbbQkbWzMybS8vIEhvbGQgb24galF1ZXJ5IRtbbQobWzMybSsbW20JG1szMm1qUXVlcnkuaG9sZFJlYWR5KCB0cnVlICk7G1ttChtbMzJtKxtbbQobWzMybSsbW20JG1szMm12YXIgcmVhZHlSZWdFeHAgPSAvXihjb21wbGV0ZXxsb2FkZWQpJC87G1ttChtbMzJtKxtbbQobWzMybSsbW20JG1szMm1mdW5jdGlvbiBhc3NldExvYWRlZCggZXZ0ICkgextbbQobWzMybSsbW20JCRtbMzJtdmFyIG5vZGUgPSBldnQuY3VycmVudFRhcmdldCB8fCBldnQuc3JjRWxlbWVudDsbW20KG1szMm0rG1ttCQkbWzMybWlmICggZXZ0LnR5cGUgPT09ICJsb2FkIiB8fCByZWFkeVJlZ0V4cC50ZXN0KCBub2RlLnJlYWR5U3RhdGUgKSApIHsbW20KG1szMm0rG1ttCQkJG1szMm1qUXVlcnkuaG9sZFJlYWR5KCBmYWxzZSApOxtbbQobWzMybSsbW20JCRtbMzJtfRtbbQobWzMybSsbW20JG1szMm19G1ttChtbMzJtKxtbbQobWzMybSsbW20JG1szMm1zZXRUaW1lb3V0KCBmdW5jdGlvbigpIHsbW20KG1szMm0rG1ttG1szMm0gICAgdmFyIHNjcmlwdCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoICJzY3JpcHQiICk7G1ttChtbMzJtKxtbbRtbMzJtICAgIHNjcmlwdC50eXBlID0gInRleHQvamF2YXNjcmlwdCI7G1ttChtbMzJtKxtbbRtbMzJtICAgIGlmICggc2NyaXB0LmFkZEV2ZW50TGlzdGVuZXIgKSB7G1ttChtbMzJtKxtbbRtbMzJtICAgICAgICBzY3JpcHQuYWRkRXZlbnRMaXN0ZW5lciggImxvYWQiLCBhc3NldExvYWRlZCwgZmFsc2UgKTsbW20KG1szMm0rG1ttG1szMm0gICAgfSBlbHNlIHsbW20KG1szMm0rG1ttG1szMm0gICAgICAgIHNjcmlwdC5hdHRhY2hFdmVudCggIm9ucmVhZHlzdGF0ZWNoYW5nZSIsIGFzc2V0TG9hZGVkICk7G1ttChtbMzJtKxtbbRtbMzJtICAgIH0bW20KG1szMm0rG1ttG1szMm0gICAgc2NyaXB0LnNyYyA9ICIuL3JlYWR5d2FpdGFzc2V0LmpzIjsbW20KG1szMm0rG1ttG1szMm0gICAgZG9jdW1lbnQuZ2V0RWxlbWVudHNCeVRhZ05hbWUoICJoZWFkIiApWyAwIF0uYXBwZW5kQ2hpbGQoIHNjcmlwdCApOxtbbQobWzMybSsbW20JG1szMm19LCAyMDAwICk7G1ttChtbMzJtKxtbbQogCWpRdWVyeShmdW5jdGlvbigpIHsbW20KG1szMW0tCQkvLyBUaGUgZGVsYXllZE1lc3NhZ2UgaXMgZGVmaW5lZCBieRtbbQobWzMxbS0JCS8vIHRoZSByZWFkeXdhaXRhc3NldC5qcyBmaWxlLCBzbyB0aGUbW20KG1szMW0tCQkvLyBuZXh0IGxpbmUgd2lsbCBvbmx5IHdvcmsgaWYgdGhpcyBET00bW20KG1szMm0rG1ttCQkbWzMybS8vIFRoZSBuZXh0IGxpbmUgd2lsbCBvbmx5IHdvcmsgaWYgdGhpcyBET00bW20KIAkJLy8gcmVhZHkgY2FsbGJhY2sgaXMgY2FsbGVkIGFmdGVyIHJlYWR5V2FpdBtbbQogCQkvLyBoYXMgYmVlbiBkZWNyZW1lbnRlZCBieSByZWFkeXdhaXRsb2FkZXIuanMbW20KIAkJLy8gSWYgYW4gZXJyb3Igb2NjdXJzLhtbbQogCQlqUXVlcnkoIiNvdXRwdXQiKS5hcHBlbmQoZGVsYXllZE1lc3NhZ2UpOxtbbQobWzMybSsbW20JCRtbMzJtc3RhcnRJZnJhbWVUZXN0KCBkZWxheWVkTWVzc2FnZSApOxtbbQogCX0pOxtbbQogCTwvc2NyaXB0PhtbbQogPC9oZWFkPhtbbQobWzE7MzNtZGlmZiAtLWdpdCB0ZXN0L2RhdGEvcmVhZHl3YWl0YXNzZXQuanMgdGVzdC9kYXRhL3JlYWR5d2FpdGFzc2V0LmpzG1ttChtbMTszM21kZWxldGVkIGZpbGUgbW9kZSAxMDA2NDQbW20KG1sxOzMzbWluZGV4IDIzMDg5NjUuLjAwMDAwMDAbW20KG1sxOzMzbS0tLSB0ZXN0L2RhdGEvcmVhZHl3YWl0YXNzZXQuanMbW20KG1sxOzMzbSsrKyAvZGV2L251bGwbW20KG1sxOzM1bUBAIC0xICswLDAgQEAbW20KG1szMW0tdmFyIGRlbGF5ZWRNZXNzYWdlID0gIkl0IHdvcmtlZCEiOxtbbQobWzE7MzNtZGlmZiAtLWdpdCB0ZXN0L2RhdGEvcmVhZHl3YWl0bG9hZGVyLmpzIHRlc3QvZGF0YS9yZWFkeXdhaXRsb2FkZXIuanMbW20KG1sxOzMzbWRlbGV0ZWQgZmlsZSBtb2RlIDEwMDY0NBtbbQobWzE7MzNtaW5kZXggOGY0YTM0NS4uMDAwMDAwMBtbbQobWzE7MzNtLS0tIHRlc3QvZGF0YS9yZWFkeXdhaXRsb2FkZXIuanMbW20KG1sxOzMzbSsrKyAvZGV2L251bGwbW20KG1sxOzM1bUBAIC0xLDI1ICswLDAgQEAbW20KG1szMW0tLy8gU2ltcGxlIHNjcmlwdCBsb2FkZXIgdGhhdCB1c2VzIGpRdWVyeS5yZWFkeVdhaXQgdmlhIGpRdWVyeS5ob2xkUmVhZHkoKRtbbQobWzMxbS0bW20KG1szMW0tLy9Ib2xkIG9uIGpRdWVyeSEbW20KG1szMW0talF1ZXJ5LmhvbGRSZWFkeSggdHJ1ZSApOxtbbQobWzMxbS0bW20KG1szMW0tdmFyIHJlYWR5UmVnRXhwID0gL14oY29tcGxldGV8bG9hZGVkKSQvOxtbbQobWzMxbS0bW20KG1szMW0tZnVuY3Rpb24gYXNzZXRMb2FkZWQoIGV2dCApIHsbW20KG1szMW0tCXZhciBub2RlID0gZXZ0LmN1cnJlbnRUYXJnZXQgfHwgZXZ0LnNyY0VsZW1lbnQ7G1ttChtbMzFtLQlpZiAoIGV2dC50eXBlID09PSAibG9hZCIgfHwgcmVhZHlSZWdFeHAudGVzdCggbm9kZS5yZWFkeVN0YXRlICkgKSB7G1ttChtbMzFtLQkJalF1ZXJ5LmhvbGRSZWFkeSggZmFsc2UgKTsbW20KG1szMW0tCX0bW20KG1szMW0tfRtbbQobWzMxbS0bW20KG1szMW0tc2V0VGltZW91dCggZnVuY3Rpb24oKSB7G1ttChtbMzFtLQl2YXIgc2NyaXB0ID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCggInNjcmlwdCIgKTsbW20KG1szMW0tCXNjcmlwdC50eXBlID0gInRleHQvamF2YXNjcmlwdCI7G1ttChtbMzFtLQlpZiAoIHNjcmlwdC5hZGRFdmVudExpc3RlbmVyICkgextbbQobWzMxbS0JCXNjcmlwdC5hZGRFdmVudExpc3RlbmVyKCAibG9hZCIsIGFzc2V0TG9hZGVkLCBmYWxzZSApOxtbbQobWzMxbS0JfSBlbHNlIHsbW20KG1szMW0tCQlzY3JpcHQuYXR0YWNoRXZlbnQoICJvbnJlYWR5c3RhdGVjaGFuZ2UiLCBhc3NldExvYWRlZCApOxtbbQobWzMxbS0JfRtbbQobWzMxbS0Jc2NyaXB0LnNyYyA9ICJkYXRhL3JlYWR5d2FpdGFzc2V0LmpzIjsbW20KG1szMW0tCWRvY3VtZW50LmdldEVsZW1lbnRzQnlUYWdOYW1lKCAiaGVhZCIgKVsgMCBdLmFwcGVuZENoaWxkKCBzY3JpcHQgKTsbW20KG1szMW0tfSwgMjAwMCApOxtbbQobWzE7MzNtZGlmZiAtLWdpdCB0ZXN0L2RhdGEvdGVzdGluaXQuanMgdGVzdC9kYXRhL3Rlc3Rpbml0LmpzG1ttChtbMTszM21pbmRleCBlZjIxMGU3Li5lMDY4NThkIDEwMDY0NBtbbQobWzE7MzNtLS0tIHRlc3QvZGF0YS90ZXN0aW5pdC5qcxtbbQobWzE7MzNtKysrIHRlc3QvZGF0YS90ZXN0aW5pdC5qcxtbbQobWzE7MzVtQEAgLTI3Niw2ICsyNzYsNyBAQBtbbSAbW210aGlzLmxvYWRUZXN0cyA9IGZ1bmN0aW9uKCkgextbbQogCQkJCSJ1bml0L2Jhc2ljLmpzIiwbW20KIBtbbQogCQkJCSJ1bml0L2NvcmUuanMiLBtbbQobWzMybSsbW20JCQkJG1szMm0idW5pdC9yZWFkeXdhaXQuanMiLBtbbQogCQkJCSJ1bml0L2NhbGxiYWNrcy5qcyIsG1ttCiAJCQkJInVuaXQvZGVmZXJyZWQuanMiLBtbbQogCQkJCSJ1bml0L2RlcHJlY2F0ZWQuanMiLBtbbQobWzE7MzNtZGlmZiAtLWdpdCB0ZXN0L3VuaXQvcmVhZHl3YWl0LmpzIHRlc3QvdW5pdC9yZWFkeXdhaXQuanMbW20KG1sxOzMzbW5ldyBmaWxlIG1vZGUgMTAwNjQ0G1ttChtbMTszM21pbmRleCAwMDAwMDAwLi5kNTcyZDEwG1ttChtbMTszM20tLS0gL2Rldi9udWxsG1ttChtbMTszM20rKysgdGVzdC91bml0L3JlYWR5d2FpdC5qcxtbbQobWzE7MzVtQEAgLTAsMCArMSwxMCBAQBtbbQobWzMybSsbW20bWzMybS8vIFNpbXBsZSBzY3JpcHQgbG9hZGVyIHRoYXQgdXNlcyBqUXVlcnkucmVhZHlXYWl0IHZpYSBqUXVlcnkuaG9sZFJlYWR5KCkbW20KG1szMm0rG1ttChtbMzJtKxtbbRtbMzJtdGVzdElmcmFtZSgbW20KG1szMm0rG1ttCRtbMzJtImhvbGRSZWFkeSB0ZXN0IG5lZWRzIHRvIGJlIGEgc3RhbmRhbG9uZSB0ZXN0IHNpbmNlIGl0IGRlYWxzIHdpdGggRE9NIHJlYWR5IiwbW20KG1szMm0rG1ttCRtbMzJtInJlYWR5d2FpdC5odG1sIiwbW20KG1szMm0rG1ttCRtbMzJtZnVuY3Rpb24oIGFzc2VydCwgalF1ZXJ5LCB3aW5kb3csIGRvY3VtZW50LCBkZWxheWVkTWVzc2FnZSApIHsbW20KG1szMm0rG1ttCQkbWzMybWFzc2VydC5leHBlY3QoIDEgKTsbW20KG1szMm0rG1ttCQkbWzMybWFzc2VydC5vayggZGVsYXllZE1lc3NhZ2UsICJJdCAiICk7G1ttChtbMzJtKxtbbQkbWzMybX0bW20KG1szMm0rG1ttG1szMm0pOxtbbQpnaXQgY29uZmlnIHBhZ2VyLmRpZmYKZ2l0IGNvbmZpZyBwYWdlci5zaG93Cg==

**Comment by scottchiefbaker** [comment]
Is it just me or is there no color in that diff until about halfway through?

**Comment by scottchiefbaker** [comment]
ping

**Comment by stevemao** [comment]
Sorry, there is color in the diff. in the base64 it prints the diff without color first then a version with color. Attaching some more images if the problem is not clear.
<img width="702" alt="screen shot 2017-02-26 at 6 05 28 pm" src="https://cloud.githubusercontent.com/assets/6316590/23337831/2f4718e8-fc4e-11e6-83bd-6bd450ce21de.png">
<img width="702" alt="screen shot 2017-02-26 at 6 04 52 pm" src="https://cloud.githubusercontent.com/assets/6316590/23337832/2f4ba3cc-fc4e-11e6-8693-6a0de840e31a.png">

**Comment by scottchiefbaker** [comment]
Looking closer at the issue here I found the "bug" is with the hunk line:

```
--- test/data/readywaitasset.js
+++ /dev/null
@@ -1 +0,0 @@
-var delayedMessage = "It worked!";
```

As far as I know that's not a valid hunk... a hunk should be of the form: `@@ -X,Y +X,Y @@` and your example is missing that first comma. Are you 100% sure this is valid output from git?

If I change that to `@@ -1,5 +0,0 @@` the text processes correctly.

**Comment by scottchiefbaker** [comment]
I think I spoke too soon. That **is** a valid hunk header if:

1. You only have one line in the file
2. You delete that one line and commit

I think I fixed it (and created a test, woohoo) in f131c709822fcd30837215423bad7fdb8b090281. Can someone verify this fix works for them, and doesn't break anything else, and then I'll land it?

**Comment by stevemao** [comment]
Had a quick look. Looks it's working :)
<img width="555" alt="screen shot 2017-03-04 at 8 56 50 am" src="https://cloud.githubusercontent.com/assets/6316590/23570508/9381797a-00b8-11e7-8dc2-ddfdc770f074.png">
